include/openmc/interpolate.h

```diff
@@ -0,0 +1,98 @@
+#ifndef OPENMC_INTERPOLATE_H
+#define OPENMC_INTERPOLATE_H
+
+#include <cmath>
+#include <vector>
+
+#include "openmc/search.h"
+
+namespace openmc {
+
+inline double interpolate_lin_lin(
+  double x0, double x1, double y0, double y1, double x)
+{
+  return y0 + (x - x0) / (x1 - x0) * (y1 - y0);
+}
+
+inline double interpolate_lin_log(
+  double x0, double x1, double y0, double y1, double x)
+{
+  return y0 + std::log(x / x0) / std::log(x1 / x0) * (y1 - y0);
+}
+
+inline double interpolate_log_lin(
+  double x0, double x1, double y0, double y1, double x)
+{
+  return y0 * std::exp((x - x0) / (x1 - x0) * std::log(y1 / y0));
+}
+
+inline double interpolate_log_log(
+  double x0, double x1, double y0, double y1, double x)
+{
+  double f = std::log(x / x0) / std::log(x1 / x0);
+  return y0 * std::exp(f * std::log(y1 / y0));
+}
+
+inline double interpolate_lagrangian(gsl::span<const double> xs,
+  gsl::span<const double> ys, int idx, double x, int order)
+{
+  Expects(order <= 3);
+  std::array<double, 4> coeffs;
+  coeffs.fill(0.0);
```

Comment:
To me it looks like we don't need an array at all -- if it's possible to just accumulate the inner product as part of the `for` loop, that would be preferred

Reply:
Yeah, you're right. Better to do it that way.